ui/dashboards/src/components/PanelDrawer/PanelEditorForm.tsx

```diff
@@ -65,10 +67,17 @@ export function PanelEditorForm(props: PanelEditorFormProps) {
     setGroupId(value);
   };
 
+  // TO DO: once we remove query editor from panel plugin, we can remove this method
+  const handlePluginSpecChange: OptionsEditorProps<UnknownSpec>['onChange'] = (spec) => {
```

Comment:
Since the query editor is part of the panel plugin, the queries are populated in plugin spec every time you update it the query editor. Thus we need to keep track of the state of the `queries`